pilot/pkg/model/push_context.go

```diff
@@ -481,36 +495,53 @@ func (ps *PushContext) GetAllSidecarScopes() map[string][]*SidecarScope {
 }
 
 // DestinationRule returns a destination rule for a service name in a given domain.
-func (ps *PushContext) DestinationRule(proxy *Proxy, hostname Hostname) *Config {
+func (ps *PushContext) DestinationRule(proxy *Proxy, service *Service) *Config {
 	// If proxy has a sidecar scope that is user supplied, then get the destination rules from the sidecar scope
 	// sidecarScope.config is nil if there is no sidecar scope for the namespace
 	// TODO: This is a temporary gate until the sidecar implementation is stable. Once its stable, remove the
 	// config != nil check
 	if proxy != nil && proxy.SidecarScope != nil && proxy.SidecarScope.Config != nil && proxy.Type == SidecarProxy {
 		// If there is a sidecar scope for this proxy, return the destination rule
 		// from the sidecar scope.
-		return proxy.SidecarScope.DestinationRule(hostname)
+		return proxy.SidecarScope.DestinationRule(service.Hostname)
 	}
 
+	// FIXME: this code should be removed once the EDS issue is fixed
```

Comment:
The fix for EDS will use DestinationRule (this function) - is this not the expectation ?

Reply:
Yes but you should supply a proxy with valid ConfigNamespace. the proxy==Nil piece exists only because of the subsetToLabels thing, which doesn't provide the namespace from which things are being accessed [i.e. the proxy's config namespace]